tasks/apply-mapping/tests/test-apply-mapping-multi-redhat-components.yaml

```diff
@@ -0,0 +1,157 @@
+---
+apiVersion: tekton.dev/v1
+kind: Pipeline
+metadata:
+  name: test-apply-mapping-multi-redhat-components
+spec:
+  description: |
+    Run the apply-mapping task with a snapshot.spec json containing multiple components 
+    and Red Hat specific repository formats. Verify that the resulting json contains the expected transformations.
+  workspaces:
+    - name: tests-workspace
+  tasks:
+    - name: setup-multi-component
+      workspaces:
+        - name: config
+          workspace: tests-workspace
+      taskSpec:
+        workspaces:
+          - name: config
+        steps:
+          - name: setup-multi-component-values
+            image: quay.io/konflux-ci/release-service-utils:e633d51cd41d73e4b3310face21bb980af7a662f
+            script: |
+              #!/usr/bin/env sh
+              set -eux
+
+              cat > $(workspaces.config.path)/test_data_multi_component.json << EOF
+              {
+                "mapping": {
+                  "components": [
+                    {
+                      "name": "comp1",
+                      "repository": "quay.io/redhat-prod/product-name----image1"
```

Comment:
My understanding is that this should no longer be supported, no? Or maybe, similar to when we were changing the tag setup, we should temporarily have the convert_to_registry function and add a block to first do that conversion if we see `quay.io/redhat-prod/` or `quay.io/redhat-pending/` on input and override the repository field and then later just remove that block including the function, once all RPAs are migrated.

Reply:
As per the discussion with Johnny, we came to the conclusion that if we drop support for the current style, merging it will be hard, so it would be nice to support both for a bit, migrate everyone, then we can remove the old style support.

Reply:
Yeah, so to make it easy to drop the support of `quay.io/redhat-*` afterwards, I think it would make sense to do it as I suggest - to first convert `quay.io/redhat-prod` to `registry.redhat.io` (or similarly for stage), so that this block can later just be removed without further refactoring.

Reply:
Okay, I am working on making/testing changes as you suggest.

Reply:
do you want me to update the test-apply-mapping-multi-redhat-components to only test registry values? because we are going to remove quay one in future after migration
Or you want me to adjust the apply-mapping task as well? with something like [this separate check](https://github.com/konflux-ci/release-service-catalog/pull/481/files#diff-89d17838cdb9d52aa345a589e39f0dacd6698104e970ee2f3bb87660e2c21f9bR218-R222).

Reply:
> do you want me to update the test-apply-mapping-multi-redhat-components to only test registry values? 

Ideally for now the tests should contain a mix of all the supported formats. Only once we remove the old format, those values should be removed from tests.

> Or you want me to adjust the apply-mapping task as well? with something like [this separate check](https://github.com/konflux-ci/release-service-catalog/pull/481/files#diff-89d17838cdb9d52aa345a589e39f0dacd6698104e970ee2f3bb87660e2c21f9bR218-R222).

I like that new temporary check. I don't think anything else is needed.

You can resolve this thread unless you want to add something here.

Reply:
Got it, Thanks!